plugins/kubernetes-backend/api-report.md

```diff
@@ -151,6 +209,8 @@ export interface RouterOptions {
   // (undocumented)
   config: Config;
   // (undocumented)
+  factory?: KubernetesFactory;
```

Comment:
Generally we'd pass in a concrete implementation here rather than a factory, do you think that would make sense? Since there is a bit of wiring to do we could get inspired by the [catalog](https://github.com/backstage/backstage/blob/d39cadd9a1dbd8fa2c51deafcf2430df7fb42e3e/packages/backend/src/plugins/catalog.ts#L25), which now has a builder that returns the actual router.

I'm thinking we should also break things down a bit so that you for example only supply a single `KubernetesServiceLocator` implementation when one is provided through code, but if we're constructing things from config we can support the set of hardcoded ones.

Reply:
I refactored the code to introduce a builder and moved the factory methods inside the builder, still allowing one to provide the services through the router options.
Also, I kept the createPlugin function to avoid any breaking change.
Tell me if you like it better!!!!